build.gradle

```diff
@@ -94,6 +94,33 @@ allprojects {
                   .exclude('**/org/apache/eventmesh/connector/openfunction/client/CallbackServiceGrpc**')
                   .exclude('**/org/apache/eventmesh/connector/jdbc/antlr**')
 
+    configurations {
+        if (!"eventmesh-starter".equals(project.name)) {
+            runtimeClasspath {
+                // Log4j API implementations
+                exclude group: 'org.apache.logging.log4j', module: 'log4j-core'
```

Comment:
There are actually a lot of executable servers under `eventmesh-connectors` besides `eventmesh-starter` that will be distributed as docker images in the future, and they run into the same problem.

![image](https://github.com/apache/eventmesh/assets/41445332/7db3373a-9e94-49fc-8010-9ea7e9ba4848)

Once we add more runnable services in the future, we will need to maintain a list here, with some inflexibility. Therefore, maybe excluding the `eventmesh-sdks` module here is enough, don't you think?

Reply:
I am not quite sure how to deal with those artifacts.

The `eventmesh-starter` artifact has **only** one possible usage: it must be started from a command line.

On the other hand artifacts like `eventmesh-runtime` or the connectors might be used by users to start EventMesh programmatically, right? In that case I wouldn't impose `log4j-core` as a dependency.

In your Docker images the logging backend can be added additionally to the "normal" dependencies, just like it is added in the `tar.gz` distribution.

If you think external projects should only use `eventmesh-sdk-java` or `eventmesh-openconnect-api`, I could move the exclusions to those projects only. What do you think?

Reply:
`eventmesh-starter` will start `eventmesh-runtime`, and the connectors can also be started from a command line.

I think external projects should only use `eventmesh-sdk-java` or any other language of SDKs under `eventmesh-sdks` (which isn't written in java). `eventmesh-openconnect-api` is less likely to be used in external projects.

Just as I said in https://github.com/apache/eventmesh/pull/4719#discussion_r1554660695, I may prefer the latter approach (first introducing logging backends normally in `eventmesh-common`, then excluding any transitive logging backend from direct SDK dependencies), because excluding logging backends globally won't resolve all problems, but also introducing more configurations on a single artifact in many scattered places, which is not that elegant. What do you think then? ☺️

I really appreciate your thoughts, but I'm sorry that I didn't explain the various uses of the project to you in the first place, which resulted in the solution you were trying to implement in the first place not being as elegant after the modifications.